jmaqs-webservices-jdk11/src/main/java/com/magenic/jmaqs/webservices/jdk11/WebServiceDriver.java

```diff
@@ -17,38 +22,140 @@
+  }
+
+  /**
+   * Sets http client.
+   * @param httpClient the http client
+   */
+  public void setHttpClient(HttpClient httpClient) {
+    this.baseHttpClient = httpClient;
+  }
+
+  /**
+   * Gets http client.
+   * @return the http client
+   */
+  public HttpClient getHttpClient() {
+    return this.baseHttpClient;
+  }
+
+  /**
+   * sets the Http Request.
+   * @param httpRequest the new http request to be set
+   */
+  public void setHttpRequest(HttpRequest httpRequest) {
+    this.baseHttpRequest = httpRequest;
+  }
+
+  /**
+   * gets the http request.
+   * @return the http request
+   */
+  public HttpRequest getHttpRequest() {
+    return this.baseHttpRequest;
+  }
+
+  /**
+   * Ensure the HTTP response was successful, if not throw a user friendly error message.
+   * @param response The HTTP response
+   */
+  private static void ensureSuccessStatusCode(HttpResponse<String> response) {
+    // Make sure a response was returned
+    if (response == null) {
+      throw new NullPointerException(HttpStatus.NO_CONTENT.toString() + " Response was null");
+    }
+
+    // Check if it was a success and if not create a user friendly error message
+    if (response.statusCode() != HttpStatus.OK.value()) {
+      throw new NotAcceptableStatusException(
+          String.format("Response did not indicate a success. %s Response code was: %s",
+              System.lineSeparator(), response.statusCode()));
+    }
+  }
+
+  /**
+   * Ensure the HTTP response has specified status, if not throw a user friendly error message.
+   * @param response The HTTP response
+   * @param expectedStatus Assert a specific status code was returned
+   */
+  private static void ensureStatusCodesMatch(HttpResponse<String> response, HttpStatus expectedStatus) {
+    // Make sure a response was returned
+    if (response == null) {
+      throw new NullPointerException(HttpStatus.NO_CONTENT.toString() + " Response was null");
+    }
+
+    // Check if it was a success and if not create a user friendly error message
+    if (response.statusCode() != expectedStatus.hashCode()) {
+      String body = response.body();
+      throw new NotAcceptableStatusException(String.format("Response status did not match expected. %s "
+              + "Response code was: %s %s Expected code was: %s %s"
+              + "Body: %s", System.lineSeparator(), response.statusCode(),
+          System.lineSeparator(), expectedStatus.hashCode(), System.lineSeparator(), body));
+    }
+  }
+
+  /**
+   * Check if the media type is supported.
+   * @param mediaType Media type to add
+   */
+  private void checkIfMediaTypeNotPresent(String mediaType) {
+    // Make sure a media type was passed in
+    if (mediaType.isEmpty()) {
+      return;
+    }
+
+    // Look for the media type
+    for (MediaType media : MediaType.values()) {
+      if (media.toString().equals(mediaType)) {
+        return;
+      }
+    }
+
+    // Add the type or throw exception???
```

Comment:
The only MAQS method I see that matches this is: 
```C#
private void AddAcceptIfNotPresent(string mediaType)
```
However we have an issue since the HttpRequest instance variable can't have headers added since after building it's values are immutable. I think this is fine for now but perhaps in the future create a method to clone the httprequest to a builder, add the header value to that builder and then reset the httpRequest instance variable on the WebServiceDriver.